app/models/concerns/taggable.rb

```diff
@@ -0,0 +1,17 @@
+module Taggable
```

Comment:
After running a few tests, I still think we should keep the `cached_tags`, it will save us a tremendous amount of queries to the database and makes stuff faster.